auth/kbase_auth_server.go

```diff
@@ -180,31 +188,39 @@ func (server *KBaseAuthServer) getUserInfo() (kbaseAuthUserInfo, error) {
 		return userInfo, err
 	}
 	err = json.Unmarshal(body, &userInfo)
-	return userInfo, err
-}
 
-// returns the username for the current KBase user accessing the
-// KBase auth server
-func (server *KBaseAuthServer) Username() (string, error) {
-	userInfo, err := server.getUserInfo()
-	return userInfo.Username, err
+	// make sure we have at least one ORCID for this user
+	if len(userInfo.Idents) < 1 {
+		return userInfo, fmt.Errorf("KBase Auth2: No providers associated with this user!")
+	}
+	foundOrcid := false
+	for _, pid := range userInfo.Idents {
+		if pid.Provider == "OrcID" {
+			foundOrcid = true
+			break
+		}
+	}
+	if !foundOrcid {
+		return userInfo, fmt.Errorf("KBase Auth2: No ORCID IDs associated with this user!")
+	}
+	return userInfo, err
 }
 
-// returns the current KBase user's registered ORCID identifiers (and/or an error)
-// a user can have 0, 1, or many associated ORCID identifiers
-func (server *KBaseAuthServer) Orcids() ([]string, error) {
-	userInfo, err := server.getUserInfo()
+// returns a normalized user info record for the current KBase user
+func (server *KBaseAuthServer) UserInfo() (UserInfo, error) {
+	kbUserInfo, err := server.kbaseUserInfo()
 	if err != nil {
-		return nil, err
+		return UserInfo{}, err
 	}
-	if len(userInfo.Idents) < 1 {
-		return nil, fmt.Errorf("No ORCID IDs associated with this user!")
+	userInfo := UserInfo{
+		Name:     kbUserInfo.Display,
+		Username: kbUserInfo.Username,
+		Email:    kbUserInfo.Email,
 	}
-	orcidIds := make([]string, 0)
-	for _, pid := range userInfo.Idents {
+	for _, pid := range kbUserInfo.Idents {
 		if pid.Provider == "OrcID" {
-			orcidIds = append(orcidIds, pid.UserName)
+			userInfo.Orcid = pid.UserName
```

Comment:
is it guaranteed that there is only ever one Orcid associated with the user?

Reply:
No, but I don't know what we would do with more than one ORCID, so I'm assuming for now that we need to select one, and that the one we select is the first one.

Reply:
What happens if the user has different OrcIDs associated with their accounts on different platforms? (well, I know what would happen -- nothing. The more useful case to think about is where they have a bunch of OrcIDs assoc'd with their KBase account, one of which is associated with their account on another platform).

Not deliberately trying to be annoying here -- just thinking of the kinds of annoying situations that may pop up in the wild.

Reply:
Makes sense! I guess my logic is that I'd like to start simple, encounter real problems, and address accordingly. At least we have separate KBase and DTS user information so we can articulate our assumptions.

Reply:
I have been burned too many times by trying to keep it simple and assuming that KBase is doing sensible things...

Reply:
I believe it! Here's an issue: https://github.com/kbase/dts/issues/81